Sources/Basics/Observability.swift

```diff
@@ -123,11 +146,19 @@ public final class ObservabilityScope: DiagnosticsEmitterProtocol, CustomStringC
         self.diagnosticsHandler.handleDiagnostic(scope: self, diagnostic: diagnostic)
     }
 
-    private struct DiagnosticsHanderWrapper: DiagnosticsHandler {
-        private let underlying: DiagnosticsHandler
+    public func emit(output: String) {
+        self.outputHandler.handleOutput(scope: self, output: output)
+    }
+
+    public func emit(step: Int64, total: Int64, unit: String?, description: String?) {
```

Comment:
Should the total be set when the scope is created?  I suppose that would make it harder to update it later.